proposals/dictionary-expressions.md

```diff
@@ -402,36 +316,152 @@ Synthesized types are free to employ any strategy they want to implement the req
 
-## Random open questions
+This is also similar to how passing to an `IEnumerable<T>` would differ from passing to some *set* type with normal collection expressions.  The target type *intentionally* affects semantics, and there is no expectation that across very different target types that one would receive the same resultant values with the same behaviors.  We do not view *dictionary types* or *key value pair elements* as changing the calculus here.
 
-### Question: Parsing ambiguity
+### Question: Allow deconstructible types?
 
-Parsing ambiguity around: `[a ? [b] : c]`
+Should we take a very restrictive view of `KeyValuePair<,>`?  Specifically, should we allow only that exact type?  Or should we allow any types with an implicit conversion to that type?  For example:
 
-Working group recommendation: Use normal parsing here.  So this would be the same as `[a ? ([b]) : (c)]` (a collection expression containing a conditional expression).  If the user wants a `key_value_pair_element` here, they can write: `[(a?[b]) : c]`
+```c#
+struct Pair<X, Y>
+{
+  public static implicit operator KeyValuePair<X, Y>(Pair<X, Y> pair) => ...;
+}
+
+Dictionary<int, string> map1 = [pair1, pair2]; // ?
+
+List<Pair<int, string>> pairs = ...;
+Dictionary<int, string> map2 = [.. pairs]; // ?
+```
+
+Similarly, instead of `KeyValuePair<,>` we could allow *any* type deconstructible to two values? For example:
 
-### Question 3
+```c#
+record struct Pair<X, Y>(X x, Y y);
 
-What are the rules when types have multiple indexers and multiple impls of `IEnumerable<KVP<,>>`
+Dictionary<int, string> map1 = [pair1, pair2]; // ?
+```
+
+Resolution: While cute, these capabilities are not needed for core scenarios to work.  They also raise concerns about where to draw the line wrt to what is the dictionary space and what is not.  As such, we will only allow `KeyValuePair<,>` for now.  And we will not do anything with tuples and/or other deconstructible types.  This is also something that could be relaxed in the future if there is sufficient feedback and motivation to warrant it.  This design space is withdrawn from dictionary expressions.
+
+### Question: Semantics when a type implements the *dictionary type* shape in multiple ways.
+
+What are the rules when types have multiple indexers and multiple impls of `IEnumerable<KVP<,>>`?  
+
+This concern already exists with *collection types*.  For those types, the rule is that we must have an *element type* as per the existing language rules.  This follows for *dictionary types*, along wih the rule that there must be a corresponding indexer for this *element type*.  If those hold, the type can be used a a *dictionary type*.  If these don't hold, it cannot be.
```

Comment:
Also retracted.  This was solved just fine for normal collection types.  Nothing  needs to change here as far as we can tell.